plugins/Live/Model.php

```diff
@@ -201,58 +201,101 @@ public function queryLogVisits($idSite, $period, $date, $segment, $countVisitors
      * @param $idSite
      * @param $lastMinutes
      * @param $segment
-     * @return array
+     * @return int
      * @throws Exception
      */
-    public function queryCounters($idSite, $lastMinutes, $segment)
+    public function getNumActions($idSite, $lastMinutes, $segment)
     {
-        $lastMinutes = (int)$lastMinutes;
+        return $this->getLastMinutesCounterForQuery(
+            $idSite,
+            $lastMinutes,
+            $segment,
+            'COUNT(*)',
+            'log_link_visit_action',
+            'log_link_visit_action.server_time >= ?'
+        );
+    }
 
-        $counters = array(
-            'visits' => 0,
-            'actions' => 0,
-            'visitors' => 0,
-            'visitsConverted' => 0,
+    /**
+     * @param $idSite
+     * @param $lastMinutes
+     * @param $segment
+     * @return int
+     * @throws Exception
+     */
+    public function getNumVisitsConverted($idSite, $lastMinutes, $segment)
+    {
+        return $this->getLastMinutesCounterForQuery(
+            $idSite,
+            $lastMinutes,
+            $segment,
+            'COUNT(*)',
+            'log_conversion',
+            'log_conversion.server_time >= ?'
         );
+    }
+
+    /**
+     * @param $idSite
+     * @param $lastMinutes
+     * @param $segment
+     * @return int
+     * @throws Exception
+     */
+    public function getNumVisits($idSite, $lastMinutes, $segment)
+    {
+        return $this->getLastMinutesCounterForQuery(
+            $idSite,
+            $lastMinutes,
+            $segment,
+            'COUNT(log_visit.visit_last_action_time)',
```

Comment:
When using `count(*)` and passing a segment eg `pageUrl` this method fails. It took me a long time to figure out what is going on. Anyway, we do not get an error when passing a `log_visit` field here and I made sure `count(log_visit.visit_last_action_time)` is as fast as `count(*)` as it still can do everything on the index and doesn't have to query the table.